pkg/config/handlerConfigurer_test.go

```diff
@@ -0,0 +1,372 @@
+				"inst2":      {"inst2", "tpml1", nil}},
+			actions: []*pb.Action{
+				{"hndlr1", []string{"repeatInst", "repeatInst"}},
+				{"hndlr1", []string{"inst2"}}},
+			want: map[string]instancesByTemplate{
+				"hndlr1": {map[string][]string{"tpml1": {"repeatInst", "inst2"}}},
+			},
+			wantError: "",
+		},
+		{
+			name:     "MultipleTemplates",
+			handlers: map[string]*HandlerBuilderInfo{"hndlr1": nil},
+			constructors: map[string]*pb.Constructor{
+				"inst1tmplA": {"inst1tmplA", "tmplA", nil},
+				"inst2tmplA": {"inst2tmplA", "tmplA", nil},
+
+				"inst3tmplB": {"inst3tmplB", "tmplB", nil},
+				"inst4tmplB": {"inst4tmplB", "tmplB", nil},
+				"inst5tmplB": {"inst5tmplB", "tmplB", nil},
+			},
+			actions: []*pb.Action{
+				{"hndlr1", []string{"inst2tmplA", "inst4tmplB", "inst5tmplB", "inst1tmplA", "inst3tmplB"}},
+			},
+			want: map[string]instancesByTemplate{
+				"hndlr1": {map[string][]string{"tmplA": {"inst2tmplA", "inst1tmplA"}, "tmplB": {"inst3tmplB", "inst5tmplB", "inst4tmplB"}}},
+			},
+			wantError: "",
+		},
+		{
+			name:     "UnionAcrossActionsWithMultipleTemplates",
+			handlers: map[string]*HandlerBuilderInfo{"hndlr1": nil, "hndlr2": nil},
+			constructors: map[string]*pb.Constructor{
+				"inst1tmplA": {"inst1tmplA", "tmplA", nil},
+				"inst2tmplA": {"inst2tmplA", "tmplA", nil},
+
+				"inst3tmplB": {"inst3tmplB", "tmplB", nil},
+				"inst4tmplB": {"inst4tmplB", "tmplB", nil},
+				"inst5tmplB": {"inst5tmplB", "tmplB", nil},
+			},
+			actions: []*pb.Action{
+				{"hndlr1", []string{"inst1tmplA", "inst3tmplB"}},
+				{"hndlr1", []string{"inst2tmplA", "inst4tmplB", "inst5tmplB"}},
+				{"hndlr2", []string{"inst2tmplA", "inst4tmplB", "inst5tmplB", "inst1tmplA", "inst3tmplB"}},
+			},
+			want: map[string]instancesByTemplate{
+				"hndlr1": {map[string][]string{"tmplA": {"inst2tmplA", "inst1tmplA"}, "tmplB": {"inst3tmplB", "inst5tmplB", "inst4tmplB"}}},
+				"hndlr2": {map[string][]string{"tmplA": {"inst2tmplA", "inst1tmplA"}, "tmplB": {"inst3tmplB", "inst5tmplB", "inst4tmplB"}}},
+			},
+			wantError: "",
+		},
+	}
+	ex, _ := expr.NewCEXLEvaluator(expr.DefaultCacheSize)
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+
+			hc := handlerConfigurer{typeChecker: ex, tmplRepo: nil}
+			v, err := hc.groupHandlerInstancesByTemplate(tt.actions, tt.constructors, tt.handlers)
+			if tt.wantError == "" {
+				if err != nil {
+					t.Errorf("got err %v\nwant <nil>", err)
+				}
+				if !deepEqualsOrderIndependent(tt.want, v) {
+					t.Errorf("got %v\nwant %v", v, tt.want)
+				}
+			} else {
+				if err == nil || !strings.Contains(err.Error(), tt.wantError) {
+					t.Errorf("got error %v\nwant %v", err, tt.wantError)
+				}
+			}
+
+		})
+	}
+
+}
+
+func deepEqualsOrderIndependent(expected map[string]instancesByTemplate, actual map[string]instancesByTemplate) bool {
+	if len(expected) != len(actual) {
+		return false
+	}
+
+	for k, exTmplCnstMap := range expected {
+		var actTmplCnstMap instancesByTemplate
```

Comment:
you are right.. must have written it to just make that stupid diff work.. didn't care there after.